gtda/diagrams/features.py

```diff
@@ -322,15 +330,21 @@ def fit(self, X, y=None):
         validate_params(self.effective_metric_params_,
                         _AVAILABLE_AMPLITUDE_METRICS[self.metric])
 
-        self.homology_dimensions_ = sorted(set(X[0, :, 2]))
+        self.homology_dimensions_ = tuple(
+            sorted([int(dim) if dim != np.inf else dim
+                    for dim in set(X[0, :, 2])])
+            )
 
         self.effective_metric_params_['samplings'], \
             self.effective_metric_params_['step_sizes'] = \
-            _bin(X, metric=self.metric, **self.effective_metric_params_)
+            _bin(X, self.metric, **self.effective_metric_params_)
 
         if self.metric == 'persistence_image':
-            self.effective_metric_params_['weights'] = \
-                _calculate_weights(X, **self.effective_metric_params_)
+            weight_function = self.effective_metric_params_.get(
+                'weight_function', None
+                )
+            if weight_function is None:
+                self.effective_metric_params_['weight_function'] = np.ones_like
```

Comment:
I missed something: if `weight_function` is not `None`, we don't use it? ( it's not passed to self.effective_metric_params_['weight_function'])

Reply:
You're right, excellent! This is the result of a bad copy-paste. I'll fix this and add a test to make sure it never happens again (I'll test the zero weight-function to check that it yields zero amplitudes and pairwise distances).

Reply:
Done in 4001abf